tools/bumper/cnao_repo_commands_test.go

```diff
@@ -339,4 +340,125 @@ var _ = Describe("Testing internal git CNAO Repo", func() {
 			expectedRemainingFiles: []string{"static", "newFile"},
 		}),
 	)
+
+	type fileInGlobListParams struct {
+		fileName       string
+		globList       []string
+		expectedResult bool
+	}
+
+	DescribeTable("fileInGlobList function",
+		func(r fileInGlobListParams) {
+			defer os.RemoveAll(gitCnaoRepo.gitRepo.localDir)
+
+			By("Running fileInGlobList on given input")
+			result := fileInGlobList(r.fileName, r.globList)
+
+			By("Checking that tree is as expected")
+			Expect(result).To(Equal(r.expectedResult), "Should match expected fileInGlobList result")
+
+		},
+		Entry("Should return false when globlist is empty", fileInGlobListParams{
+			fileName:       "",
+			globList:       []string{},
+			expectedResult: false,
+		}),
+		Entry("Should return false when globlist doesn't match file string", fileInGlobListParams{
+			fileName:       "fileName",
+			globList:       []string{"some", "other", "glob", "file"},
+			expectedResult: false,
+		}),
+		Entry("Should return true when globlist has glob match to file string", fileInGlobListParams{
+			fileName:       "fileName",
+			globList:       []string{"other", "file*"},
+			expectedResult: true,
+		}),
+		Entry("Should return true when globlist has exact match to file string", fileInGlobListParams{
+			fileName:       "fileName",
+			globList:       []string{"other", "fileName", "file"},
+			expectedResult: true,
+		}),
+	)
+
+	type collectModifiedToTreeListParams struct {
+		files                 []string
+		allowedList           []string
+		expectedTreeEntryList []*github.TreeEntry
+		shouldReturnErr       bool
+	}
+
+	DescribeTable("collectModifiedToTreeList function",
+		func(r collectModifiedToTreeListParams) {
+			defer os.RemoveAll(gitCnaoRepo.gitRepo.localDir)
+
+			By("Modifying files in the Repo")
+			modifyFiles(gitCnaoRepo.gitRepo.localDir, r.files)
+
+			By("Running collectModifiedToTreeList api")
+			treeList, err := gitCnaoRepo.collectModifiedToTreeList(r.allowedList)
+			if r.shouldReturnErr {
+				Expect(err).To(HaveOccurred(), "should fail collecting Modified files to tree list")
+			} else {
+				Expect(err).ToNot(HaveOccurred(), "should succeed collecting Modified files to tree list")
+			}
+
+			By("Checking that tree is as expected")
+			Expect(treeList).To(ConsistOf(r.expectedTreeEntryList), "tree list should be equal to expected")
+
+		},
+		Entry("Should return error when no file is modified, allowedList allows all files", collectModifiedToTreeListParams{
+			files:                 []string{},
+			allowedList:           []string{"*"},
+			expectedTreeEntryList: nil,
+			shouldReturnErr:       true,
+		}),
```

Comment:
Are you sure we have to return error at this scenario ?

Reply:
I think so, otherwise the bump pr will be empty. We should at least have a change in components.yaml,

Reply:
What I mean is that instead of return an error here we can return an empty `TreeEntry` and  the caller check that if this is empty do nothing.

Reply:
I agree it's more fluent that way. Also, it's shouldn't stop other bumps so I change that as well.
Done